src/libraries/System.IO.FileSystem.AccessControl/tests/TempAclDirectory.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Security.AccessControl;
+using System.Security.Principal;
+
+namespace System.IO
+{
+    /// <summary>
+    /// Represents a temporary directory.
+    /// Disposing will recurse all files and directories inside it, ensure the
+    /// appropriate access control is set, then delete all of them.
+    /// </summary>
+    public sealed class TempAclDirectory : TempDirectory
+    {
+        protected override void DeleteDirectory()
+        {
+            try
+            {
+                var identity = new SecurityIdentifier(WellKnownSidType.BuiltinUsersSid, null);
+                var accessRule = new FileSystemAccessRule(identity, FileSystemRights.FullControl, AccessControlType.Allow);
+
+                foreach (string dirPath in Directory.EnumerateDirectories(Path, "*", SearchOption.AllDirectories))
+                {
+                    var dirInfo = new DirectoryInfo(dirPath);
+                    dirInfo.SetAccessControl(new DirectorySecurity(dirPath, AccessControlSections.Access));
+                }
+
+                foreach (string filePath in Directory.EnumerateFiles(Path, "*", SearchOption.AllDirectories))
+                {
+                    var fileInfo = new FileInfo(filePath);
+                    fileInfo.SetAccessControl(new FileSecurity(filePath, AccessControlSections.Access));
+                }
+
+                var rootDirInfo = new DirectoryInfo(Path);
+                rootDirInfo.SetAccessControl(new DirectorySecurity(Path, AccessControlSections.Access));
```

Comment:
would it be possible to instead of searching for all folders and files apply the setting recursively for the `Path` directory and then just call `base.DeleteDirectory` which would take care of the removal?

Reply:
I think you're describing the original behavior of the parent class I am overriding:

https://github.com/dotnet/runtime/blob/b23ccde65bff6b0964371cc47967af162ec5f2df/src/libraries/Common/tests/System/IO/TempDirectory.cs#L43-L47